### PR TITLE
Provide initial implementation of SIA-R81

### PR DESCRIPTION
**Pull request by kasperisager**
This still lacks this bit from the programmatic link context:

> [...] or in a table header cell that is associated with the table cell that contains the link.

### CODE REVIEW

File: packages/alfa-rules/src/sia-r81/rule.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import { isIgnored } from "../common/predicate/is-ignored";
+
+import { Question } from "../common/question";
+
+const { isElement, hasNamespace, hasId } = Element;
+const { map, flatMap, isEmpty } = Iterable;
+const { and, or, not, equals } = Predicate;
+
+export default Rule.Atomic.of<Page, Iterable<Element>, Question>({
+  uri: "https://siteimprove.github.io/sanshikan/rules/sia-r41.html",
```
Comment (Jym77) [inline packages/alfa-rules/src/sia-r81/rule.ts:29]:
```suggestion
  uri: "https://siteimprove.github.io/sanshikan/rules/sia-r81.html",
```

**Comment by kasperisager** [comment]
Good catch! 🤦

File: packages/alfa-rules/src/sia-r81/rule.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        const elements = document
+          .descendants({ flattened: true, nested: true })
+          .filter(
+            and(
+              isElement,
+              and(
+                hasNamespace(Namespace.HTML, Namespace.SVG),
+                hasRole(
+                  or(Role.hasName("link"), (role) =>
+                    role.inheritsFrom(Role.hasName("link"))
```
Comment (Jym77) [inline packages/alfa-rules/src/sia-r81/rule.ts:42]: Here also we might want an "inclusive inherits" helper as it happens quite a few times.

File: packages/alfa-rules/src/sia-r81/rule.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                  groups
+                    .get(node.name())
+                    .getOrElse(() => List.empty<Element>())
+                    .append(element)
+                );
+              }
+
+              return groups;
+            }, Map.empty<Option<string>, List<Element>>())
+            .filter((elements) => elements.size > 1)
```
Comment (Jym77) [inline packages/alfa-rules/src/sia-r81/rule.ts:70]: I see this one we do the size check 😁

**Pull request merged**